include/seqan3/argument_parser/validators.hpp

```diff
@@ -373,23 +373,36 @@ class file_validator_base
             throw validation_error{detail::to_string("The given filename ", path.string(), " has no extension. Expected"
                                                      " one of the following valid extensions:", extensions, "!")};
 
-        // Drop the dot.
-        std::string drop_less_ext = path.extension().string().substr(1);
+        std::string const path_as_string{path.filename().string()};
+        std::string_view file_path{path_as_string};
+
+        // Leading dot indicates a hidden file is not part of the extension.
+        if (file_path.front() == '.')
+            file_path = file_path.substr(1);
+
+        // Store a string_view containing all extensions to 
+        std::string_view const all_extensions{file_path.substr(file_path.find(".") + 1)};
 
         // Compares the extensions in lower case.
-        auto case_insensitive_equal_to = [&] (std::string const & ext)
+        auto case_insensitive_ends_with = [&] (std::string const & ext)
         {
-            return std::ranges::equal(ext, drop_less_ext, [] (char const chr1, char const chr2)
-                   {
-                       return std::tolower(chr1) == std::tolower(chr2);
-                   });
+            return string_ends_with(file_path, ext);
         };
 
+        bool ext_found{false};
+
+        // Iterate over all possible extensions, starting with the longest. E.g., "foo.bar.baz", "bar.baz", "baz".
+        for (size_t pos = file_path.find("."); !ext_found && pos != std::string_view::npos; pos = file_path.find("."))
```

Comment:
What happens here? Why not just iterate over the valid extensions and check each time if `file_path` ends with that?

Reply:
Hah, yeah no idea. Probably the amalgamation of different ideas...